package.json

```diff
@@ -467,38 +470,99 @@
         },
         "debuggers": [
             {
-                "type": "docker",
-                "label": "Docker: Node.js",
+                "type": "docker-launch",
+                "label": "Docker: Launch",
                 "configurationSnippets": [
                     {
-                        "label": "Docker: Attach to Node",
```

Comment:
>Attach to Node [](start = 42, length = 14)

We need to keep the Attach to Node functionality, though?

Reply:
I could go either way; the snippet is so basic and makes so many assumptions about the environment in which its used that I'm not sure how much benefit it has.  That said, there's no real reason we *have* to remove it right now.

Reply:
This snippet is a little odd. It's provided by us, but is strictly a Node Attach thing, i.e. we don't actually do the work of supporting it. I'll leave it in for now since it's probably harmless, but we should consider removing it in the future.

Reply:
If we're keeping this snippet, probably should keep the equivalent one provided by the dummy configuration provider.  That said, there's no real reason for us to keep/register that configuration provider; maybe we should fold that into our own.  We could even rename it `'docker'` as I'm thinking now that `'docker-launch'` somewhat conflates the idea that a debugger may operate in both `'launch'` and `'attach'` modes.  In the case of .NET Core, the configuration provider uses `'launch'` under the covers, but, in the case of Node.js, the configuration provider uses `'attach'` under the covers.  Technically, the configuration provider's not really "launching" Docker; the container is now being build/spun-up by the pre-launch tasks.

Reply:
Good point; I'd forgotten about the dummy debug config provider.

I'd rather not try to fold together that and this; I'm concerned there's too much risk of regressions to existing configs and breaking future work.